File: ui/lib/api/traits.ts
/**
 * Traits API client.
 */

import { BaseApiClient } from "./base";
import type { TraitVocabularyResponse } from "../types";

export class TraitsApi extends BaseApiClient {
  /**
   * Get the trait vocabulary.
   */
  async getTraitVocabulary(): Promise<TraitVocabularyResponse> {
    return this.get<TraitVocabularyResponse>("/api/v1/traits");
  }
}

File: src/voiceobs/server/routes/traits.py
"""Trait vocabulary routes."""

from fastapi import APIRouter

from voiceobs.server.models.response import TraitVocabularyResponse
from voiceobs.server.services.scenario_generation.trait_vocabulary import TRAIT_VOCABULARY

router = APIRouter(prefix="/api/v1/traits", tags=["Traits"])


@router.get(
    "",
    response_model=TraitVocabularyResponse,
    summary="Get trait vocabulary",
    description="Get the available persona traits organized by category.",
)
async def get_trait_vocabulary() -> TraitVocabularyResponse:
    """Get the trait vocabulary for persona creation."""
    return TraitVocabularyResponse(vocabulary=TRAIT_VOCABULARY)
